---
layout: post
title:  "Visualization of a Spleen Tissue and is Identification"
author: Yuqi Zhang
jhed: yzhan354
categories: [ HW5 ]
image: homework/hw5/yuqiz.png
featured: false
---

We are given a spatial protein expression dataset identify its originality. The figures shown above should be labeled A to G from left to right and from top to bottom. 

**Figure A. Scree Plot of Variation Explained by Different Number of PCs.** This plot is a scree plot of the variation that could be explained by different number of PCs. This helps choose the number of PC to be used for clustering. 

**Figure B. Elbow Plot of Within Group Sum of Squares of Different K Choice for K means clustering.** This plot help explains the variation explained within each cluster for different choice of K in the algorithm. This helps choose the final K. 

**Figure C. Spatial Distribution of Cell Clusters with K means Clustering.** From Figure B, we choose K = 13 to perform K means clustering. This image uses position to visualize the spatial distribution of cells in the x, y coordinate and uses the color hue visual channel to encode the cluster of cells. 

**Figure D. Average Protein Expression **

To process the data, first we normalize protein expression with total protein expression across cells. 
PCA helps reduce dimension and remove correlations when we later use it to cluster the cells. Based on the scree plot of PCA, we choose first 9 PCs to be used for later clustering because the first 9 PCs explain most of the variations, and the scree plot's line doesn't change much after 9. 

Then we want to choose the number k for kmeans clustering. The elbow plot of Within Group Sum of Squares help us choose k for kmeans clustering algorithm to cluster cells. It explain the variation within clusters for different choice of k, although it is lower the better, in the plot, since the variation explained doesn't drop much after K = 13, we choose this as our K for K means clustering. 

```R
library(ggplot2)
library(Rtsne)
library(gridExtra)
library(scattermore)
visium_data = read.csv("~/genomic-data-visualization/data/Visium_Cortex_varnorm.csv.gz", 
                       row.names = 1)
visium_data_gene = visium_data[, c(3:ncol(visium_data))]
# Normalization: 
visium_data_gene_normal = visium_data_gene/rowSums(visium_data_gene)*1e6
visium_data_gene_normal = log10(visium_data_gene_normal + 1)

g1 = "Olig1"
g2 = "Olig2"
g3 = "Pdgfra"
group_list = c()
found = FALSE
k = 20

# Find optimal cluster that has differential expression of marker genes
while(!found)
{
  k_clusters = kmeans(visium_data_gene_normal, centers = k)
  visium_data_gene_normal$cluster = as.factor(k_clusters$cluster)
  for (i in 1:k)
  {
    group = visium_data_gene_normal$cluster == i
    test_stat_g1 = wilcox.test(visium_data_gene_normal[group, g1], 
                               visium_data_gene_normal[!group, g1], alternative = "greater")
    test_stat_g2 = wilcox.test(visium_data_gene_normal[group, g2], 
                               visium_data_gene_normal[!group, g2], alternative = "greater")
    test_stat_g3 = wilcox.test(visium_data_gene_normal[group, g3], 
                               visium_data_gene_normal[!group, g3], alternative = "less")
    if (test_stat_g1$p.value < 0.01 & test_stat_g2$p.value < 0.01 & test_stat_g3$p.value < 0.01)
      group_list = append(group_list, i)
  }
  if (length(group_list) == 1)
  {
    found = TRUE
  }
  else 
    k = k - 1
  if (k < 2)
  {
    print("Cannot find a group. ")
    break
  }
  print(k)
}

cells_interest = visium_data_gene_normal$cluster == group_list[1]

pos_df = visium_data[, 1:2]

# Visuailze Spatial Distribution
p1 = ggplot(pos_df, aes(x = x, y = y)) + 
  geom_scattermore(aes(col = cells_interest), pointsize = 2) + theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  labs(color = "Is Mature Oligodendrocyte")


df <- reshape2::melt(
  data.frame(id=rownames(visium_data_gene_normal), 
             visium_data_gene_normal[, c('Olig1', 'Olig2', 'Pdgfra', 'Mog', 'Mbp', 'Sox10')], 
             col=as.factor(cells_interest)))

# Visualize Marker Genes
p2 = ggplot(data = df, 
       mapping = aes(x=col, y=value, fill=col)) + 
  geom_violin() + 
  theme_classic() + 
  facet_wrap(~ variable) + ggtitle("Marker Genes Violin Plot") +
  ylab("Gene Expression") + xlab("") + labs(color = "Is Mature Oligodendrocyte")

png("~/genomic-data-visualization/homework/hw4/yuqiz.png", width = 1200, height = 500)
grid.arrange(p1, p2, ncol = 2, 
             top = "Identify Mature Oligodendrocyte in Visium Dataset with Marker Genes")
dev.off()
```
