---
layout: page
title:  "Lesson 2: Spatially Resolved Transcriptomic Data"
---

# Lecture 2

## 2.0 Lesson learning objectives

By the end of this lesson, we should understand what is spatially resolved transcriptomic data, how the data is generated, and how we can begin visualizing and interacting with the data.

## 2.1 Why spatially resolved transcriptomics?


## 2.2 Low-throughput approaches for spatially resolved transcriptomic profiling

### smFISH

### microdissection

## 2.3 High-throughput approaches for spatially resolved transcriptomic profiling

### Spatially resolved highly multiplxed transcriptomic imaging with MERFISH

MERFISH is an imaging method capable of simultaneously measuring the copy number and spatial distribution of hundreds to thousands of RNA species in fixed cells. This technique was introduced in the paper *Chen et al. Spatially resolved, highly multiplexed RNA profiling in single cells. Science (2015)*

### Spatially resolved transcriptomic sequencing with Visium

Visium is a commercial platform that uses glass slides with arrayed oligonucleotides containing positional barcodes to generate cDNA libraries with positional information for RNA sequencing. The technique was introduced in the paper *Stahl et al. Visualization and analysis of gene expression in tissue sections by spatial transcriptomics. Science (2016)* and later licensed to 10X Genomics for commercialization.

---

# Hands-on component 2

Our in-class hands-on component will downloading and analyzing either a MERFISH or Visium dataset of a coronal section of the mouse brain.

---

# Class Lesson Notes 2

Notes from class will be uploaded here.

---

# Homework Assignment 2

Read the original Chen et al and Stahl et al papers. Quiz next week may incorporate questions from these readings.

Complete Homework Assignment 0 if you registered late. 

